t/tasks/inventory/generic/ipmi/fru_memories.t

```diff
@@ -0,0 +1,1398 @@
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM3',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 12,
+            'SERIALNUMBER'     => '00002009',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM4',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 13,
+            'SERIALNUMBER'     => '00006009',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM5',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 14,
+            'SERIALNUMBER'     => '00007010',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM6',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 15,
+            'SERIALNUMBER'     => '00001067',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        },
+        {
+            'CAPACITY'         => '2048',
+            'CAPTION'          => 'CPU1_DIMM7',
+            'DESCRIPTION'      => 'DIMM',
+            'MANUFACTURER'     => 'SK Hynix',
+            'MEMORYCORRECTION' => 'Single-bit ECC',
+            'MODEL'            => 'HYMP525P72CP4-Y5',
+            'NUMSLOTS'         => 16,
+            'SERIALNUMBER'     => '04008004',
+            'SPEED'            => '533',
+            'TYPE'             => 'DDR2'
+        }
+    ]
+);
+
+plan tests => 3 * (scalar keys %tests) + 1;
+
+foreach my $test (keys %tests) {
+    clearFruCache();
+
+    my $dmidecode = "resources/generic/dmidecode/$test";
+    my $fru = "resources/generic/ipmitool/fru/$test";
+    my $inventory = FusionInventory::Test::Inventory->new();
+
+    lives_ok {
+        FusionInventory::Agent::Task::Inventory::Generic::Dmidecode::Memory::doInventory(
+            inventory => $inventory,
+            file      => $dmidecode
+        );
+    } "test $test: dmidecode/memory doInventory()";
```

Comment:
Even if it's easier to test dmidecode/memory here, this is not this test script purpose. You should even just reproduce the Dmidecode/Memory.pm doInventory() to prepare your inventory. So you even not have to modify it to accept file in %params. So just replace with the following code to emulate dmidecode inventory:
```suggestion
    my $memories = FusionInventory::Agent::Task::Inventory::Generic::Dmidecode::Memory::_getMemories(file => $dmidecode);
    foreach my $memory (@$memories) {
        $inventory->addEntry(
            section => 'MEMORIES',
            entry   => $memory
        );
    }
```
Btw, you may want to add your dmidecode output to the related dmidecode test script so your outputs are validated at the right place.